### DISCUSSION TITLE
report_id and second audit

### DISCUSSION THREAD

**Discussion by AaronDoty** [GSA-TTS/FAC#2722] [Q&A]
Hello, not sure if this is the correct forum to ask about the "general" endpoint.
If an auditee submits a second audit and the fac_accepted_date is changed, will a new report_id be generated?
Or will the report_id remain the same and the new audit PDF file will be associated with the it?
Also, can the report_id be considered a primary key, or maybe the report_id coupled with the fac_accepted_date?
Thanks for any assistance provided.

**Comment by jadudm** [comment]
Hi @AaronDoty , 

You're asking a question that edges towards resubmissions, which we're handling fully at the start of the new year.

At the moment, we will see two audits. 

We will be implementing a full resubmission process for tracking resubmissions properly. However, the design is in flux at the moment. As a principle, we do not want to edit prior records. But, how to do this so it does not disrupt agency operations is tricky.

I know we won't have a compound key. However, we may end up with some ancillary tables (e.g. a `versions` or `revisions` table) that will help users rebuild a version history. We will likely add a `version` field to the `general` endpoint. However, *this is speculative* at this point, as the model is not finalized.

We will document this fully as we develop the design with agency partners.
